src/parse.rs

```diff
@@ -37,18 +43,227 @@ impl Expr {
+                let mut value_map = HashMap::new();
+                // Used for the ExprSource. See ExprSource::AttrSet for
+                // details on why we create both a hashmap and a vector.
+                let mut definitions = vec![];
+
+                for entry in set.entries() {
+                    // Where x, y, z are KeyValuePairs:
+                    //
+                    //   services.bluetooth.enable = true;
+                    //                      +------------+ x
+                    //            +----------------------+ y
+                    //   +-------------------------------+ z
+                    //
+                    // Hovering over `x` should show `true`.
+                    // Hovering over `y` should show `{ enable }`.
+                    // Hovering over `z` should show `{ bluetooth }`.
+                    //
+                    // This matches what we would see for verbose syntax:
+                    //
+                    //   services = { bluetooth = { enable = true; }; };
+                    //
+                    // So, we rewrite paths into the verbose syntax.
+
+                    let mut path = entry
+                        .key()
+                        .ok_or(ERR_PARSING)?
+                        .path()
+                        .map(|node| Self::parse(node, scope.clone()).map(Gc::new))
+                        .collect::<Vec<_>>();
+
+                    // NOTE: This pops from the end, so we want to remove
+                    //       the inmost element before reversing
+                    let inmost_key = path.pop().unwrap()?;
+
+                    // After this, our path lists path elements from right to left
+                    path.reverse();
+
+                    let inmost_value_syntax = entry.value().ok_or(ERR_PARSING)?;
+                    let entry_end = inmost_value_syntax.text_range().end();
+                    let inmost_value = Self::parse(inmost_value_syntax, new_scope.clone())?;
+
+                    let here_start = inmost_key.range.ok_or(ERR_PARSING)?.start();
+
+                    let mut cursor_range = TextRange::new(here_start, entry_end);
+                    let mut cursor_key_name = inmost_key.as_ident()?;
+                    let mut cursor_value = Gc::new(Expr {
+                        value: GcCell::new(None),
+                        source: ExprSource::KeyValuePair {
+                            key: Ok(inmost_key),
+                            value: Ok(Gc::new(inmost_value)),
+                        },
+                        range: Some(cursor_range),
+                        scope: new_scope.clone(),
+                    });
+
+                    for element in path {
+                        let here_start = element.as_ref()?.range.ok_or(ERR_PARSING)?.start();
+
+                        // Create an invisible attr set
+                        let tmp_map = NixValue::Map(HashMap::from_iter([(
```

Comment:
Did you miss this during your merge? I actually replaced `[` with `vec![`, otherwise it wouldn't compile locally.